src/com/goide/runconfig/testing/frameworks/gobench/GobenchRunConfigurationProducer.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2013-2016 Sergey Ignatov, Alexander Zolotov, Florin Patan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.goide.runconfig.testing.frameworks.gobench;
+
+import com.goide.psi.GoFunctionOrMethodDeclaration;
+import com.goide.runconfig.testing.GoTestFinder;
+import com.goide.runconfig.testing.GoTestRunConfigurationProducerBase;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.util.PsiTreeUtil;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class GobenchRunConfigurationProducer extends GoTestRunConfigurationProducerBase implements Cloneable {
+  public GobenchRunConfigurationProducer() {
+    super(GobenchFramework.INSTANCE);
+  }
+
+  @NotNull
+  @Override
+  protected String getPackageConfigurationName(@NotNull String packageName) {
+    return "gobench package '" + packageName + "'";
+  }
+
+  @NotNull
+  @Override
+  protected String getFileConfigurationName(@NotNull String fileName) {
+    return "gobench " + super.getFileConfigurationName(fileName);
+  }
+
+  @Nullable
+  @Override
+  protected GoFunctionOrMethodDeclaration findTestFunctionInContext(@NotNull PsiElement contextElement) {
+    GoFunctionOrMethodDeclaration function = PsiTreeUtil.getNonStrictParentOfType(contextElement, GoFunctionOrMethodDeclaration.class);
```

Comment:
I think that GoMethodDeclaration is not appliable for `gobench`, so something like 

```
type T int

function (T) B<caret>enchmarkName() {
}
```

shouldn't produce `gobench` run configuration, right? If so then it's better to replace with

```
GoFunctionDeclaration function = PsiTreeUtil.getNonStrictParentOfType(contextElement, GoFunctionDeclaration.class);
```